pkg/apis/extensions/types.go

```diff
@@ -268,6 +268,11 @@ const (
 
 	// Replace the old RCs by new one using rolling update i.e gradually scale down the old RCs and scale up the new one.
 	RollingUpdateDeploymentStrategyType DeploymentStrategyType = "RollingUpdate"
+
+	// The Deployment controller is not going to orchestrate ReplicaSets for
+	// Deployments that use this strategy. Useful for delegating orchestration of
+	// certain Deployments to external controllers.
+	NoopDeploymentStrategyType DeploymentStrategyType = "Noop"
```

Comment:
What is an example of delegating orchestration to external controllers(do you mean thirty party extensions) ?  Would there be an argument for doing that for replica set also ?